apps/backend/src/modules/categories/category.repository.ts

```diff
@@ -117,6 +117,37 @@ class CategoryRepository implements Repository {
 		);
 	}
 
+	public async findByIds(categoryIds: number[]): Promise<CategoryEntity[]> {
+		const categories = await this.categoryModel
+			.query()
+			.whereIn("id", categoryIds)
+			.select("*");
+
+		return await Promise.all(
+			categories.map(async (category) => {
+				const scoresModel = await this.categoryModel
+					.query()
+					.from(DatabaseTableName.QUIZ_SCORES)
+					.where({ categoryId: category.id })
+					.castTo<CategoryModel[]>();
+
+				const scoreEntities = scoresModel.map((score) => {
+					return CategoryEntity.initializeNew({
```

Comment:
I believe this is not score, this is category

Reply:
I don't think that we need these scores here to be honest, output of the endpoints for getAll and getByIds will be used for getting all categories to display them in the form component made by Iqbal and to retake the quiz. In both cases, we just need the category ids essentialy (and names for the form)

Reply:
We don't need scores. You named a variable `scoreEntities`, but using `CategoryEntity.initializeNew`. It should be `categories`

Reply:
This approach is used in other methods of this repository, for example the one in findAll, i though that this is to be expected